### DISCUSSION TITLE
Unavailable for your site because your domain is not properly configured to support HTTPS

### DISCUSSION THREAD

**Discussion by D3rJust1n** [community/community#134087] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I have a problem with setting my custom domain. I get this message and I don't know what the problem is. 
I have my domain at Cloudflare. 
If anyone here in the community has a tip, I would be very grateful 🙏🏼 
Below is a picture

![image](https://github.com/user-attachments/assets/16c7cbed-0818-4171-90ab-aa33b5c9fdce)

**Comment by dante-barreda** [comment]
The issue you're encountering is related to the HTTPS enforcement on your custom domain configured through GitHub Pages. Even though your DNS settings are correct (as indicated by "DNS check successful"), the "Enforce HTTPS" option is unavailable because your domain is not properly configured to support HTTPS.

1.- Check SSL/TLS Settings in Cloudflare. Ensure that your SSL/TLS encryption mode is set to Full or Full (Strict). These modes ensure that Cloudflare connects to your origin server using HTTPS.

2.- Configure SSL Certificate. Ensure that you have an SSL certificate installed on your server if using Full (Strict) mode. If you don’t have one, you can generate a free SSL certificate from Let’s Encrypt.

3.- Make sure you have added the correct CNAME record in Cloudflare pointing to your GitHub Pages site.

DNS changes can take some time to propagate. Ensure you wait for a few minutes up to a few hours and then check again if HTTPS can be enforced.

Try enabling the "Enforce HTTPS" option again.

Hope this helps

I'll give you a link to some documentation
https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site

**Comment by uvindu94** [comment]
It seems like there might be an issue with the HTTPS configuration for your custom domain. Here are a few steps to troubleshoot and resolve this:

1. Verify DNS Settings:

Ensure that your DNS settings at Cloudflare are correctly configured. You should have a CNAME record pointing www.d3rjustin.xyz to d3rjustin.github.io and an A record pointing d3rjustin.xyz to the GitHub Pages IP addresses:

        185.199.108.153
        185.199.109.153
        185.199.110.153
        185.199.111.153

2. Check SSL/TLS Settings on Cloudflare:
  
        Go to the SSL/TLS settings on your Cloudflare dashboard.

        Ensure that the SSL/TLS encryption mode is set to "Full" or "Full (strict)". The "Flexible" mode can cause issues with GitHub Pages.
        
3. Enable Universal SSL on Cloudflare:

      In the SSL/TLS settings, make sure that Universal SSL is enabled. This will allow Cloudflare to provide an SSL certificate for your domain.
      
4. Verify GitHub Pages Settings:
      
      In your GitHub repository settings, under the "Pages" section, make sure that your custom domain is correctly set.
      Ensure that "Enforce HTTPS" is enabled, if the option becomes available after setting up your domain correctly on Cloudflare.

5. Allow Time for DNS Propagation:

      DNS changes can take some time to propagate. If you've recently updated your DNS settings, it might take a few hours for the changes to take effect globally.

6. Clear Browser Cache:

    Sometimes, cached data can cause issues. Clear your browser cache or try accessing your site in incognito mode.

7. Check for Errors in Cloudflare's Dashboard:

    Cloudflare sometimes provides detailed error messages or warnings in their dashboard. Check if there are any alerts related to your domain's SSL/TLS configuration.

If you've checked all these steps and the issue persists, please provide more details about your current DNS records and SSL/TLS settings for me. i would like to help you!
